lib/ansible/playbook/taggable.py

```diff
@@ -53,8 +53,11 @@ def evaluate_tags(self, only_tags, skip_tags, all_vars):
             for tag in tags:
                 if isinstance(tag, list):
                     _temp_tags.update(tag)
-                else:
+                elif isinstance(tag, string_types):
```

Comment:
it should really always be a list by the time it gets here ... I'm thinking alternative patch:

```

diff --git a/lib/ansible/playbook/playbook_include.py b/lib/ansible/playbook/playbook_include.py
index ec3909e38e..461e84dc34 100644
--- a/lib/ansible/playbook/playbook_include.py
+++ b/lib/ansible/playbook/playbook_include.py
@@ -20,8 +20,8 @@ from __future__ import (absolute_import, division, print_function)
 __metaclass__ = type

 import os
-
 from ansible.errors import AnsibleParserError, AnsibleAssertionError
+from ansible.module_utils.common.collections import is_sequence
 from ansible.module_utils.six import iteritems, string_types
 from ansible.parsing.splitter import split_args, parse_kv
 from ansible.parsing.yaml.objects import AnsibleBaseYAMLObject, AnsibleMapping
@@ -83,7 +83,12 @@ class PlaybookInclude(Base, Conditional, Taggable):
             temp_vars.update(new_obj.vars)
             param_tags = temp_vars.pop('tags', None)
             if param_tags is not None:
-                entry.tags.extend(param_tags.split(','))
+                if isinstance(param_tags, string_types):
+                    entry.tags.extend(param_tags.split(','))
+                elif is_sequence(param_tags):
+                    entry.tags.extend(param_tags)
+                else:
+                    raise AnsibleParserError('Invalid type supplied for tags keyword, expected list but got %s' % type(param_tags))
             entry.vars = temp_vars
             entry.tags = list(set(entry.tags).union(new_obj.tags))
             if entry._included_path is None:```

Reply:
tags is a list but, the tags list item may be a list as well
~~~yaml
- debug: msg=test
  tags:
  - [ one, two, three ]
~~~

Reply:
so the check should really be 'list of strings' ... with a flatten for backwards compat